airsenal/framework/optimization_transfers.py

```diff
@@ -120,27 +159,31 @@ def make_optimum_double_transfer(
     best_score = -1
     best_squad = None
     best_pid_out, best_pid_in = [], []
+    # We will order the list of potential subs via the sum of expected points
+    # over a specified range of gameweeks.
     ordered_player_lists = {
         pos: get_predicted_points(
             gameweek=gameweek_range, position=pos, tag=tag, season=season
         )
         for pos in ["GK", "DEF", "MID", "FWD"]
     }
-    for i in range(len(squad.players) - 1):
+    if not candidate_players_to_remove:
+        candidate_players_to_remove = squad.players
+    for i in range(len(candidate_players_to_remove) - 1):
         positions_needed = []
-        pout_1 = squad.players[i]
+        pout_1 = candidate_players_to_remove[i]
 
         new_squad_remove_1 = fastcopy(squad)
         new_squad_remove_1.remove_player(pout_1.player_id, gameweek=transfer_gw)
-        for j in range(i + 1, len(squad.players)):
+        for j in range(i + 1, len(candidate_players_to_remove)):
```

Comment:
I don't think we'd want the 2nd player out to also be limited to `candidate_players_to_remove`? E.g. if we have 4 City players the best way to solve it might be to make 2 transfers where only 1 of the players transferred out is a City player.

It could get messy but I think we want the first `len(candidate_players_to_remove)` players out to be limited to players in `candidate_players_to_remove`, and any subsequent players out to be any player in `squad.players`?

Reply:
ah there's some additional logic later on that I missed here, having another think...

Reply:
Yep good point!   I just simplified the logic in `make_best_transfers` and changed the logic in `make_optimum_double_transfer` now, so that:
* It will now do 0, 1, or 2 transfers, as per the strategy.
* If we are doing 1 transfer, and we have a non-empty candidate_players_to_remove list, `make_optimum_single_transfer` will loop over that list (as before).
* If we are doing 2 transfers, we also pass over num_compulsory_transfers (the number of players we need to remove) to `make_optimum_double_transfer`.   If that number is 1, we only use `candidate_players_to_remove` for the outer loop, and still loop over the whole squad in the inner loop.   If it is 2 or more, we use `candidate_players_to_remove` for both the inner and outer loops.
* The double-loop logic is a bit more clunky now for making sure we only try each pair of players once (but I think it was plain wrong in the last commit!), but I think it's ok now, for when we loop over the whole squad in both loops (i.e. the normal situation).  We actually are trying pairs twice if we're using `candidate_players_to_remove` in both the inner and outer loops, but the combinatorics are so much smaller in that case that I think it won't slow us down (plus it's a super-rare situation!)